apps/browser/qml/pages/components/TabView.qml

```diff
@@ -40,131 +34,248 @@ SilicaGridView {
     signal closeAllCanceled
     signal closeAllPending
 
-    onCountChanged: if (count > 0) closingAllTabs = false
-    onClosingAllTabsChanged: if (closingAllTabs) closeAllPending()
-
-    width: parent.width - Theme.horizontalPageMargin
-    height: parent.height
-    x: Theme.horizontalPageMargin
-    currentIndex: -1
-    header: PageHeader {
-        //: Tabs
-        //% "Tabs"
-        title: qsTrId("sailfish_browser-he-tabs")
-    }
-    footer: spacer
-    cellHeight: thumbnailHeight + Theme.paddingLarge
-    cellWidth: thumbnailWidth + Theme.paddingLarge
+    property var _remorsePopup
 
-    delegate: TabItem {
-        id: tabItem
+    // TODO: Delete height calculation for Pulley Menu after adding popup menu
+    property real _pulleyMenuApparentHeight
+    readonly property int _pulleyMenuHeight: 7 * Theme.pixelRatio
+    function _changeMenuApparentHeight(contentY) {
+        _pulleyMenuApparentHeight = Math.min(Math.round(_pulleyMenuHeight - tabs.tabBarHeight - contentY), _pulleyMenuHeight)
+    }
 
-        enabled: !closingAllTabs
-        opacity: enabled ? 1.0 : 0.0
-        Behavior on opacity { FadeAnimator {}}
+    anchors.fill: parent
 
-        width: thumbnailWidth
-        height: thumbnailHeight
+    Private.TabView {
+        id: tabs
 
-        GridView.onAdd: AddAnimation {
-            target: tabItem
+        property int headerHeight
+        anchors {
+            fill: parent
+            bottomMargin: tabsToolBar.height
         }
-        GridView.onRemove: RemoveAnimation {
-            target: tabItem
+
+        header: Private.TabBar {
+            id: headerTabs
+            model: headerModel
+
+            Rectangle {
+                anchors {
+                    fill: parent
+                    topMargin: _pulleyMenuApparentHeight
```

Comment:
This should now be safe:
```suggestion
                    topMargin: _pulleyMenuHeight
```
Which would allow you to remove the `_changeMenuApparentHeight()` logic I think.

Reply:
I agree, removed